nemo/collections/nlp/modules/common/text_generation_strategy.py

```diff
@@ -349,14 +349,21 @@ def neva_process_prompts(prompt, tokenizer, multimodal_cfg, num_media_latents, c
             'conversations': [{'from': 'User', 'value': prompt}, {'from': 'Assistant', 'value': '',},],
         }
 
-        for turn in record['conversations']:  #
+        for turn in record['conversations']:
             if turn.get('value') is not None:
                 turn['value'] = re.sub('<image>', f'{DEFAULT_IMAGE_TOKEN}\n', turn['value'])
         list_data_dict.append(record)
 
-        sources = preprocess_multimodal(
-            copy.deepcopy(list_data_dict), multimodal_cfg, num_media_latents
-        )  # HARDCODED FOR NOW
+        num_media_latents = min((num_media_latents // 14) * (num_media_latents // 14), 576)
```

Comment:
this line is because for video neva, the actual media tensors shape from current model is 256 x 256, so we need some logic to make `model_config.data.image_token_len` from 224  to 256. However, if we do not hard coded 576 here, it will break neva v1.5 model inference as it uses `image_token_len = 576`. Would appreciate some suggestions to deal with this more elegantly.

Reply:
[resolved]